DashWallet/Sources/UI/CrowdNode/Portal/CrowdNodePortalViewController.swift

```diff
@@ -15,10 +15,105 @@
 //  limitations under the License.
 //
 
+import Combine
+
+// MARK: - CrowdNodePortalController
+
 final class CrowdNodePortalController: UIViewController {
+    private let viewModel = CrowdNodeModel.shared
+    private var cancellableBag = Set<AnyCancellable>()
+
+    @IBOutlet var depositInput: UITextField!
+    @IBOutlet var withdrawInput: UITextField!
+    @IBOutlet var outputLabel: UILabel!
+    @IBOutlet var addressLabel: UILabel!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureLayout()
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        cancellableBag.removeAll()
+    }
+
+    @objc func copyAddress() {
+        UIPasteboard.general.string = addressLabel.text
+    }
+
+    @IBAction func deposit() {
+        guard let inputText = depositInput.text else { return }
+        let dash = DSPriceManager.sharedInstance().amount(forDashString: inputText)
+
+        Task {
+            do {
+                try await viewModel.deposit(amount: dash)
+                dismiss(animated: true, completion: nil)
```

Comment:
@Syn-McJ Should we wrap this line into:

```
await MainActor.run {
  dismiss(animated: true, completion: nil)
}
```

Reply:
@tikhop pretty sure the execution context should be returned back to the main thread after `await` automatically since the view controller is already a MainActor and `Task` inherits the context.
At least, I'm not seeing any threading errors when `dismiss` is run.